packages/viz/src/lib/Layer.ts

```diff
@@ -42,32 +50,31 @@ export class Layer {
     });
   }
 
-  public async getDeckGLLayer(layerProps: { layerType?: any } = {}) {
+  public async getDeckGLLayer() {
     // TODO: Parse through Babel
-    const { layerType: sublayerType, ...overridenStyleProps} = layerProps;
-    const { urlTemplates } = await this._layerInstantiation.then(this._parseInstantiationResult);
-
-    const defaultLayerProps = {
-      getLineColor: [192, 0, 0],
-      getFillColor: [200, 120, 80],
-      lineWidthMinPixels: 1,
-      pointRadiusMinPixels: 5,
-      urlTemplates,
-      uniquePropertyName: 'cartodb_id'
-    };
-
-    const layerProperties = Object.assign({}, defaultLayerProps, overridenStyleProps);
+    const {urlTemplates, geometryType} = await this._layerInstantiation.then(this._parseInstantiationResult);
+    const defaultGeometryStyles = defaultStyles[geometryType];
+
+    const layerProperties = Object.assign(
+      { urlTemplates },
+      defaultGeometryStyles.getProperties(),
+      this._layerStyles.getProperties()
+    );
+
     return new MVTTileLayer(layerProperties);
   }
 
   private _parseInstantiationResult(instantiationData: any) {
-    const urlData = instantiationData.metadata.url.vector;
+    const metadata = instantiationData.metadata;
 
+    const urlData = metadata.url.vector;
     const urlTemplates = urlData.subdomains.map(
       (subdomain: string) => urlData.urlTemplate.replace('{s}', subdomain)
     );
 
-    return { urlTemplates };
+    const geometryType = metadata.layers[0].meta.stats.geometryType.split('ST_')[1];
+
+    return { urlTemplates, geometryType };
   }
```

Comment:
Should we abstract all this parsing and map instantiation to another class/whatsoever within our visualization module?

Reply:
I would say this corresponds to maps package, who knows the internal about map instantiation and the metadata fields

Reply:
I am gonna write it down in Clubhouse as well, because this refactor will add a lot of changed lines in this PR.